internal/app/middleware.go

```diff
@@ -0,0 +1,23 @@
+package app
+
+import (
+	"github.com/gin-gonic/gin"
+)
+
+func ErrorHandle() gin.HandlerFunc {
```

Comment:
i think better to make like this:
```
var (
	// ErrInternal is an error for when an internal service fails to process the request
	ErrInternal = errors.New("internal error")
	// ErrDataNotFound is an error for when requested data is not found
	ErrDataNotFound = errors.New("data not found")
	// ErrNoUpdatedData is an error for when no data is provided to update
	ErrNoUpdatedData = errors.New("no data to update")
	// ErrConflictingData is an error for when data conflicts with existing data
	ErrConflictingData = errors.New("data conflicts with existing data in unique column")
	// ErrInvalidCredentials is an error for when the credentials are invalid
	ErrInvalidCredentials = errors.New("invalid email or password")
	// ErrUnauthorized is an error for when the user is unauthorized
	ErrUnauthorized = errors.New("user is unauthorized to access the resource")
)

var errorStatusMap = map[error]int{
        ErrInternal:                   http.StatusInternalServerError,
        ErrDataNotFound:               http.StatusNotFound,
        ErrConflictingData:            http.StatusConflict,
        ErrInvalidCredentials:         http.StatusUnauthorized,
        ErrUnauthorized:               http.StatusUnauthorized,
}

func HandleError(ctx *gin.Context, err error) {
	statusCode, ok := errorStatusMap[err]
	if !ok {
		statusCode = http.StatusInternalServerError
	}

	errRsp := errorResponse(errMsg)
	ctx.JSON(statusCode, errRsp)
}

func errorResponse(errMsgs []string) errorResponse {
	return errorResponse{
		Success:  false,
		Messages: errMsgs,
	}
}
```

Reply:
Why not use https://developer.mozilla.org/docs/Web/HTTP/Status ?

Reply:
Yes, the error file will be soon.